### ISSUE TITLE
[replay] Failed to bind Wayland shell

### ISSUE THREAD

**Issue by liamwhite** [LunarG/gfxreconstruct#1245]
**Describe the replay bug:**
When attempting to play back any capture of an application that uses Wayland WSI on GNOME 44 using Wayland, I get this console output every time:
```
[gfxrecon] ERROR - Failed to bind Wayland shell
gfxrecon-replay: source/gfxreconstruct/framework/decode/vulkan_replay_consumer_base.cpp:1887: VkResult gfxrecon::decode::VulkanReplayConsumerBase::CreateSurface(gfxrecon::decode::InstanceInfo*, const std::string&, VkFlags, gfxrecon::decode::HandlePointerDecoder<VkSurfaceKHR_T*>*): Assertion `window_factory' failed.
Aborted (core dumped)
```

**Verify before submission:**
* Was trimming enabled?
   N/A
* Was replayer renamed if necessary?
   N/A
* Was --sync used if title is known to need forced synchronization?
   N/A

**Build Environment:**
commit 06e7639323af3218160d3a35a4b57b5d398f3ea3 (HEAD -> dev, origin/dev, origin/HEAD)

**To Reproduce**
1. Capture any Vulkan program using Wayland WSI
2. Run gfxrecon-replay on the generated capture file

**Screenshots:**
N/A

**System environment:**
 - Capture environment: Mesa 23.1.6, RX 6700 XT (RADV), Fedora Linux 6.4.12-200.fc38.x86_64
 - Replay environment: Mesa 23.1.6, RX 6700 XT (RADV), Fedora Linux 6.4.12-200.fc38.x86_64

**Title configuration:**
  - N/A
  - 1.0.0
  - Steam ID: N/A
  - Is DXR enabled? No

**Additional information (optional):**
  - is there a SHA for which replayer is known to replay correctly?  N/A
  - Is there an older trace that works?  What SHA was used to build those capture DLLs?  N/A
  - Does a newer capture work?  N/A
  - Does the capture file replay correctly on a different GPU? N/A

**Comment by marius-pelegrin-arm** [comment]
If the wayland version is too recent, there's no wl_shell anymore and wayland instead use XDG-Shell.

I implemented a support for it in #1190, maybe you can try it and tell me if the issue is solved ?

**Comment by charles-lunarg** [comment]
Tested locally using Gnome 46 with the merged changes, and is now fixed.
